kobotoolbox/kpi
# fix(NLP): NLP features broken on non-Stripe instances TASK-1322

### 🗒️ Checklist

1. [x] run linter locally
2. [x] update all related docs (API, README, inline, etc.), if any
3. [x] draft PR with a title `<type>(<scope>)<!>: <title> TASK-1234`
4. [x] tag PR: at least `frontend` or `backend` unless it's global
5. [x] fill in the template below and delete template comments
6. [x] review thyself: read the diff and repro the preview as written
7. [x] open PR & confirm that CI passes
8. [x] request reviewers, if needed
9. [ ] delete this section before merging

### 📣 Summary
Fix bug in NLP feature when stripe is disabled

### 📖 Description
This PR fixes a bug found on the NLP feature that occurred when stripe was disabled. Some code was refactored and a new unit tests was added to ensure that no issue arises when stripe is disabled. 

### 💭 Notes
Ensure that all the necessary env variables are set up. While testing this on the local instance, I found that the variable GS_BUCKET_NAME was not set.

### 👀 Preview steps
1. Disable stripe by changing the docker file `docker-compose.frontend.custom.yml` on kobo-docker
2. Run the set up script `./run.py --setup` to apply the docker file changes.
3. Ensure that stripe is not enabled by going to the admin page and make sure that the djstripe admin screens are not available.
4. Go to an asset with an audio file question 
5. You can use an existing submission or submit a new one with a short audio file
6. Transcribe and translate the audio file
7. Make sure there are no errors in the API responses when attempting to process the audio file
